compiler/packages/babel-plugin-react-compiler/src/ReactiveScopes/InferReactiveScopeVariables.ts

```diff
@@ -379,6 +379,14 @@ export function findDisjointMutableValues(
              */
             operand.identifier.mutableRange.start > 0
           ) {
+            if (
+              instr.value.kind === 'FunctionExpression' ||
+              instr.value.kind === 'ObjectMethod'
+            ) {
+              if (operand.identifier.type.kind === 'Primitive') {
+                continue;
+              }
+            }
```

Comment:
see comment above, i'd like to understand this more. why this exact combination and not primitives of other instruction types, for example.

Reply:
(Let's continue discussion here https://github.com/facebook/react/pull/32096/files#r1956773709)